src/enact/function_wrappers.py

```diff
@@ -192,28 +299,113 @@ def create(cls, function_name: str, fun: Callable) -> 'Call':
     call._callable = fun
     return call
 
-  def call(self, resource: CallArgs) -> interfaces.ResourceBase:
+  def call(self, resource: CallArgs) -> Any:
     """Call the wrapped function."""
     if not self._callable:
       raise invocations.InvocationError(
         'Cannot repeat a call of an unregistered function. '
         'Please make sure your top-level function is registered.')
     args, kwargs = resource.to_python_args()
-    result = self._callable(*args, **kwargs)
-    return resource_registry.wrap(result)
+    return self._callable(*args, **kwargs)
+
+@resource_registry.register
+@dataclasses.dataclass
+class AsyncCall(invocations.AsyncInvokable[CallArgs, Any]):
```

Comment:
I guess there's no concept of a private constructor in Python? That would be the classic way to implement this factory pattern kinda thing. It seems strange that you can initialize an AsyncCall without a callable, but not sure there is a way around it.

Reply:
Not that I'm aware of.